connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1217,7 +1217,8 @@ class SparkConnectPlanner(
     rel.getReadTypeCase match {
       case proto.Read.ReadTypeCase.NAMED_TABLE =>
         val multipartIdentifier =
-          CatalystSqlParser.parseMultipartIdentifier(rel.getNamedTable.getUnparsedIdentifier)
+          session.sessionState.sqlParser
+            .parseMultipartIdentifier(rel.getNamedTable.getUnparsedIdentifier)
```

Comment:
I think we can add a test under `SparkConnectPlannerSuite` that uses a cloned `SparkSession` that defines a separate parser with `SparkSession.withExtensions` that overrides `parseMultipartIdentifier` which throws an exception (see `MyParser` at `SparkSessionExtensionSuite`).

Reply:
thanks, I will have a try

Reply:
done